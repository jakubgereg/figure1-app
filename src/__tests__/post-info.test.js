import React from "react";
import ReactDOM from "react-dom";
import PostInfo from "../components/post-info";
import { shallow } from "enzyme";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<PostInfo />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders views correctly", () => {
  const wrapper = shallow(<PostInfo viewCount="20" />);
  const should = <span className="views">20</span>;
  expect(wrapper.contains(should)).toEqual(true);
});

it("renders a snapshot", () => {
  const wrapper = shallow(
    <PostInfo viewCount={20} starCount={1} followCount={4} />
  );
  expect(wrapper).toMatchSnapshot();
});
