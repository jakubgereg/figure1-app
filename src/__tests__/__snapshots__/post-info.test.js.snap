// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostInfo
    followCount={4}
    starCount={1}
    viewCount={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="star link"
        >
          1
        </span>,
        <span
          className="views"
        >
          20
        </span>,
        <span
          className="follow"
        >
          4
        </span>,
      ],
      "className": "links",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "star link",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": 1,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 20,
          "className": "views",
        },
        "ref": null,
        "rendered": 20,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 4,
          "className": "follow",
        },
        "ref": null,
        "rendered": 4,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="star link"
          >
            1
          </span>,
          <span
            className="views"
          >
            20
          </span>,
          <span
            className="follow"
          >
            4
          </span>,
        ],
        "className": "links",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "star link",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": 1,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 20,
            "className": "views",
          },
          "ref": null,
          "rendered": 20,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 4,
            "className": "follow",
          },
          "ref": null,
          "rendered": 4,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
